### DISCUSSION TITLE
In glue Schema, Partition key expects similar Data type for all partitions, to pass set expression in GetPartitions api ?

### DISCUSSION THREAD

**Discussion by Sach1nAgarwal** [aws/aws-sdk-cpp#2417] [Q&A]
After running the glue crawler, it generates schema, when all partition_key in the generated schema have same data type, 
then  get_partitions api after set_expression passes, 
If any one partition_key have different data type in schema, then after setting set_expression, get_partitions api throwing an error "**InvalidInputException**" 

Is this expected behavior or not?

**Comment by Sach1nAgarwal** [comment]
If let suppose two partitions are there in schema:- 


Column name  |  Data Type 

       c1         int
       c2        float
 
 GetPartitions fails, if setting expression (  c1 >  20 ) .   
 
 Now if change data type of c2 to int or bigint or char,  get partition passes.

**Reply by jmklix** [reply]
I'm still a little confused about what you are asking. Can you provide more details and maybe a code snipet that gives you the "invalideInputException"

You can also check out GetPartition details [here](https://docs.aws.amazon.com/glue/latest/webapi/API_GetPartition.html)

**Reply by Sach1nAgarwal** [reply]
@jmklix 
It is [GetPartitions](https://docs.aws.amazon.com/glue/latest/webapi/API_GetPartitions.html#API_GetPartitions_RequestSyntax). 
Corrected the heading.

![Screenshot from 2023-04-02 13-51-18](https://user-images.githubusercontent.com/15383869/229341180-2278bb0a-4e0e-41a7-8ad8-63794c35d67d.png)

If we do set_expression( tinyint > 20) then GetPartitions give  "invalidInputException",  
Now if I **change Data Type** of **Column name float, from float to int**, then GetPartitions gives correct results.
